File: Lista/Go/Q8.go
package main

import "fmt"

type Empregado struct {
    Nome   string
    Cargo  string
    Salario float64
}

func (e Empregado) ExibirDetalhes() {
    fmt.Printf("Nome: %s, Cargo: %s, Salário: R$%.2f\n", e.Nome, e.Cargo, e.Salario)
}

type Empresa struct {
    Nome       string
    Empregados []Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado Empregado) {
    e.Empregados = append(e.Empregados, empregado)
}

func (e Empresa) ExibirEmpregados() {
    fmt.Printf("Empresa: %s\n", e.Nome)
    fmt.Println("Lista de empregados:")
    for _, empregado := range e.Empregados {
        empregado.ExibirDetalhes()
    }
}

func (e Empresa) CalcularTotalSalarios() float64 {
    total := 0.0
    for _, empregado := range e.Empregados {
        total += empregado.Salario
    }
    return total
}

func main() {
    empresa := Empresa{
        Nome: "Tech Solutions",
    }

    empregado1 := Empregado{Nome: "Alice", Cargo: "Desenvolvedora", Salario: 5000.00}
    empregado2 := Empregado{Nome: "Bob", Cargo: "Analista de Sistemas", Salario: 4500.00}
    empregado3 := Empregado{Nome: "Carlos", Cargo: "Gerente de Projetos", Salario: 7000.00}

    empresa.AdicionarEmpregado(empregado1)
    empresa.AdicionarEmpregado(empregado2)
    empresa.AdicionarEmpregado(empregado3)

    empresa.ExibirEmpregados()

    totalSalarios := empresa.CalcularTotalSalarios()
    fmt.Printf("Total de salários pagos pela empresa: R$%.2f\n", totalSalarios)
}

File: Lista/Python/Q8.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

    def exibir_detalhes(self):
        return f'Nome: {self.nome}, Cargo: {self.cargo}, Salário: R${self.salario:.2f}'

class Empresa:
    def __init__(self, nome_empresa):
        self.nome_empresa = nome_empresa
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def exibir_empregados(self):
        print(f'Empregados da {self.nome_empresa}:')
        for empregado in self.empregados:
            print(empregado.exibir_detalhes())

empregado1 = Empregado('João', 'Desenvolvedor', 5000)
empregado2 = Empregado('Ana', 'Gerente de Projetos', 8000)
empregado3 = Empregado('Pedro', 'Designer', 4000)

empresa = Empresa('Tech Solutions')

empresa.adicionar_empregado(empregado1)
empresa.adicionar_empregado(empregado2)
empresa.adicionar_empregado(empregado3)

empresa.exibir_empregados()